test/route_end_to_end_connectivity_test.cc

```diff
@@ -0,0 +1,291 @@
+  const api::RoadPosition end_position{lane_1_0_1, api::LanePosition{kLaneLength, 0., 0.}};
+  const api::LaneSRange lane_s_range_1(kLane_1_0_1, api::SRange(0, kLaneLength));
+  const api::LaneSRange lane_s_range_m1(kLane_1_0_m1, api::SRange(0, kLaneLength));
+  const routing::Phase phase(0 /*index*/, kLinearTolerance, {start_position_1, start_position_m1}, {end_position},
+                             {lane_s_range_m1, lane_s_range_1}, road_network_.get());
+  const routing::Route dut({phase}, road_network_.get());
+
+  const std::vector<std::string> errors = dut.ValidateEndToEndConnectivity();
+
+  ASSERT_FALSE(errors.empty()) << SerializeErrors(errors);
+}
+
+TEST_F(SinglePhaseTest, SinglePhaseWithTwoLanesAndEndPositionsIsNotConnectedEndToEnd) {
+  const api::RoadPosition start_position{lane_1_0_1, api::LanePosition{0., 0., 0.}};
+  const api::RoadPosition end_position_1{lane_1_0_1, api::LanePosition{kLaneLength, 0., 0.}};
+  const api::RoadPosition end_position_m1{lane_1_0_m1, api::LanePosition{kLaneLength, 0., 0.}};
+  const api::LaneSRange lane_s_range_1(kLane_1_0_1, api::SRange(0, kLaneLength));
+  const api::LaneSRange lane_s_range_m1(kLane_1_0_m1, api::SRange(0, kLaneLength));
+  const routing::Phase phase(0 /*index*/, kLinearTolerance, {start_position}, {end_position_1, end_position_m1},
+                             {lane_s_range_m1, lane_s_range_1}, road_network_.get());
+  const routing::Route dut({phase}, road_network_.get());
+
+  const std::vector<std::string> errors = dut.ValidateEndToEndConnectivity();
+
+  ASSERT_FALSE(errors.empty()) << SerializeErrors(errors);
+}
+
+class MultiPhaseTest : public MultipleStraightSegmentsTest {};
+
+TEST_F(MultiPhaseTest, MultiPhaseWithOneLaneEachIsConnectedEndToEnd) {
+  const api::RoadPosition first_start_position{lane_1_0_1, api::LanePosition{0., 0., 0.}};
+  const api::RoadPosition first_end_position{lane_1_0_1, api::LanePosition{kLaneLength, 0., 0.}};
+  const api::LaneSRange first_lane_s_range(kLane_1_0_1, api::SRange(0, kLaneLength));
+  const routing::Phase first_phase(0 /*index*/, kLinearTolerance, {first_start_position}, {first_end_position},
+                                   {first_lane_s_range}, road_network_.get());
+  const api::RoadPosition last_start_position{lane_2_0_1, api::LanePosition{0., 0., 0.}};
+  const api::RoadPosition last_end_position{lane_2_0_1, api::LanePosition{kLaneLength, 0., 0.}};
+  const api::LaneSRange last_lane_s_range(kLane_2_0_1, api::SRange(0, kLaneLength));
+  const routing::Phase last_phase(1 /*index*/, kLinearTolerance, {last_start_position}, {last_end_position},
+                                  {last_lane_s_range}, road_network_.get());
+  const routing::Route dut({first_phase, last_phase}, road_network_.get());
+
+  const std::vector<std::string> errors = dut.ValidateEndToEndConnectivity();
+
+  ASSERT_TRUE(errors.empty()) << SerializeErrors(errors);
+}
+
+TEST_F(MultiPhaseTest, MultiPhaseWithOneLaneEachIsNotGeometricallyConnectedEndToEnd) {
```

Comment:
Ditto. Add a comment explaining why having phases with one lane each results in the route not being connected end-to-end.

(Ditto similar situations below.)

Reply:
[5415ba7](https://github.com/maliput/maliput_integration_tests/pull/81/commits/5415ba7b4a80d037e1a4fe16999f64b6bfd528aa)